config/config_test.go

```diff
@@ -14,6 +14,105 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestExternalCacheURLValidate(t *testing.T) {
+	type testOut struct {
+		expectError  bool
+		expectedHost string
+		expectedPath string
+	}
+
+	testCases := []struct {
+		desc string
+		data ExternalCache
+		out  testOut
+	}{
+		{
+			"With http://",
+			ExternalCache{Host: "http://www.google.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/path/v1"},
```

Comment:
Why do we specify `expectedHost` and `expectedPath` when we except an error? Neither of those are checked. Consider removing in all test cases where `expectError` is `true`.

Reply:
They get checked at the end if we don't expect error but with the recent changes in the function they are definitely redundant. Removed